src/ias_client/src/ias_client.rs

```diff
@@ -0,0 +1,230 @@
+    spid_cert: Vec<u8>,
+    // Timeout for the client requests in seconds
+    timeout: Duration,
+}
+
+const SIGRL_LINK: &str = "/attestation/sgx/v2/sigrl";
+const AVR_LINK: &str = "/attestation/sgx/v2/report";
+// Note: Structure can be used for serialization and deserialization, but it won't skip null values
+const ISV_ENCLAVE_QUOTE: &str = "isvEnclaveQuote";
+const PSE_MANIFEST: &str = "pseManifest";
+const NONCE: &str = "nonce";
+
+/// Implement how the IasClient is going to be used
+impl IasClient {
+    /// default constructor for IasClient, remember to use setters later
+    pub fn default() -> Self {
+        IasClient {
+            ias_url: String::new(),
+            spid_cert: [].to_vec(),
+            timeout: Duration::new(300, 0),
+        }
+    }
+
+    /// constructor for IasClient
+    pub fn new(url: String, cert: Vec<u8>, time: Option<u64>) -> Self {
+        IasClient {
+            ias_url: url,
+            spid_cert: cert,
+            timeout: Duration::new(time.unwrap_or(300), 0),
+        }
+    }
+
+    /// Setters for IasClient structure
+    pub fn set_ias_url(&mut self, url: String) {
+        self.ias_url = url;
+    }
+
+    pub fn set_spid_cert(&mut self, cert: Vec<u8>) {
+        self.spid_cert = cert;
+    }
+
+    pub fn set_timeout(&mut self, timeout: Duration) {
+        self.timeout = timeout;
+    }
+
+    /// Get request to receive signature revocation list for input Group ID (gid). Accepts
+    /// optional 'gid' and optional 'api_path' as input. Optional 'gid' field is used for the
+    /// case of IAS Proxy server, which receives request with 'gid' appended already.
+    ///
+    /// return: A ClientResponse object containing the following:
+    ///     Body of the response has 'signature revocation list', the body of the response from ISA.
+    ///     Header of the response has nothing.
+    pub fn get_signature_revocation_list(
+        &self,
+        gid: Option<&str>,
+        api_path: Option<&str>,
+    ) -> Result<ClientResponse, ClientError> {
+
+        // Path to get SigRL from
+        let mut final_path = String::new();
+        final_path.push_str(self.ias_url.as_str());
+        // Received REST path if any
+        let received_path = match api_path {
+            Some(path_present) => path_present,
+            _ => SIGRL_LINK,
+        };
+        final_path.push_str(received_path);
+        // Append gid to the path if present
+        let received_gid = match gid {
+            Some(gid_present) => gid_present,
+            _ => "",
+        };
+        final_path.push_str(received_gid);
+        let url = final_path.parse::<Uri>().expect("Error constructing URI from string");
+        debug!("Fetching SigRL from: {}", url);
+
+        // Send request to get SigRL
+        let client = get_client(self.spid_cert.as_slice())
+            .expect("Eror creating http/s client");
+        // TODO: Add logic for request timeout
+        let response = client.get(url);
```

Comment:
Can we rename response to responsefut
Is this blocking call ? Any default timeout ?

Reply:
hyper doesn't have a runtime associated with the future, so no comments on default timeout. We have to associate with the runner instance which in our case the blocking function call read_future_response() has it.

I will rename the variable.